In [1]:
import pandas as pd
import matplotlib.pyplot as plt

pd.set_option('display.max_columns', 20)

In [2]:
census_path = 'data/san-diego-census-2017/'

census_df = pd.read_csv(census_path + 'San Diego.csv')
meta_df = pd.read_csv(census_path + 'metaindicator.csv')

print(f"Census Data Shape: {census_df.shape}")
print(f"Metadata Shape:    {meta_df.shape}")

print("\n--- Column Dictionary (first 10) ---")
display(meta_df.head(10))

print("\n--- Census Data Preview ---")
display(census_df.head())

Census Data Shape: (628, 21)
Metadata Shape:    (13, 5)

--- Column Dictionary (first 10) ---


,Unnamed: 0,var_id,label,concept,short_name
0,0,B25077_001E,Estimate!!Median value (dollars),MEDIAN VALUE (DOLLARS),median_house_value
1,1,B02001_002E,Estimate!!Total!!White alone,RACE,total_pop_white
2,2,B01003_001E,Estimate!!Total,TOTAL POPULATION,total_pop
3,3,B25003_003E,Estimate!!Total!!Renter occupied,TENURE,total_rented
4,4,B25001_001E,Estimate!!Total,HOUSING UNITS,total_housing_units
5,5,B09019_006E,Estimate!!Total!!In households!!In family hous...,HOUSEHOLD TYPE (INCLUDING LIVING ALONE) BY REL...,hh_female
6,6,B09019_001E,Estimate!!Total,HOUSEHOLD TYPE (INCLUDING LIVING ALONE) BY REL...,hh_total
7,7,B15003_002E,Estimate!!Total!!No schooling completed,EDUCATIONAL ATTAINMENT FOR THE POPULATION 25 Y...,total_bachelor
8,8,B25018_001E,Estimate!!Median number of rooms,MEDIAN NUMBER OF ROOMS,median_no_rooms
9,9,B19083_001E,Estimate!!Gini Index,GINI INDEX OF INCOME INEQUALITY,income_gini



--- Census Data Preview ---


,Unnamed: 0,GEOID,geometry,median_age,total_pop,total_pop_white,tt_work,hh_total,hh_female,total_bachelor,...,income_gini,total_housing_units,total_rented,median_no_rooms,median_house_value,NAME,state,county,tract,sub_30
0,0,6073007701,"POLYGON ((-13052014.93 3867748.53, -13051974.6...",31.9,4074.0,3737.0,2243.0,4074.0,301.0,0.0,...,0.4182,2394.0,1499.0,3.9,852000.0,"Census Tract 77.01, San Diego County, California",6,73,7701,False
1,1,6073008002,"POLYGON ((-13052206.51 3870188.93, -13052193.6...",37.0,2968.0,2677.0,1558.0,2968.0,296.0,0.0,...,0.4094,1117.0,345.0,5.5,893300.0,"Census Tract 80.02, San Diego County, California",6,73,8002,False
2,2,6073007903,"POLYGON ((-13051964.72 3869215.2, -13051923.76...",30.2,4660.0,4079.0,3100.0,4660.0,425.0,0.0,...,0.4528,2582.0,1713.0,3.8,423500.0,"Census Tract 79.03, San Diego County, California",6,73,7903,False
3,3,6073008364,"POLYGON ((-13050548.18 3875343.81, -13050522.3...",30.1,5811.0,3135.0,3169.0,5811.0,448.0,0.0,...,0.3986,2902.0,2090.0,3.7,601100.0,"Census Tract 83.64, San Diego County, California",6,73,8364,False
4,4,6073008305,"POLYGON ((-13051530.24 3879241.24, -13051529.9...",27.1,2002.0,1053.0,879.0,2002.0,152.0,0.0,...,0.4797,815.0,706.0,4.0,517900.0,"Census Tract 83.05, San Diego County, California",6,73,8305,False


In [3]:
print("--- Census Data Info ---")
census_df.info()

print("\n--- Missing Values (top 10 columns) ---")
missing = census_df.isnull().sum().sort_values(ascending=False)
print(missing[missing > 0].head(10))

print("\n--- Statistical Summary ---")
display(census_df.describe())

print("\n--- Possible geographic ID columns ---")
id_candidates = [col for col in census_df.columns if any(
    key in col.lower() for key in ['geo', 'tract', 'fips', 'id'])]
print(id_candidates)

--- Census Data Info ---
<class 'pandas.DataFrame'>
RangeIndex: 628 entries, 0 to 627
Data columns (total 21 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           628 non-null    int64  
 1   GEOID                628 non-null    int64  
 2   geometry             628 non-null    str    
 3   median_age           628 non-null    float64
 4   total_pop            628 non-null    float64
 5   total_pop_white      628 non-null    float64
 6   tt_work              628 non-null    float64
 7   hh_total             628 non-null    float64
 8   hh_female            628 non-null    float64
 9   total_bachelor       628 non-null    float64
 10  median_hh_income     628 non-null    float64
 11  income_gini          628 non-null    float64
 12  total_housing_units  628 non-null    float64
 13  total_rented         628 non-null    float64
 14  median_no_rooms      628 non-null    float64
 15  median_house_value   628 n

,Unnamed: 0,GEOID,median_age,total_pop,total_pop_white,tt_work,hh_total,hh_female,total_bachelor,median_hh_income,income_gini,total_housing_units,total_rented,median_no_rooms,median_house_value,state,county,tract
count,628.000000,6.280000e+02,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,628.000000,6.280000e+02,628.0,628.0,628.000000
mean,313.500000,6.073014e+09,37.212620,5228.765924,3702.920382,2332.433121,5228.765924,547.920382,66.525478,75402.798712,0.410741,1904.248408,832.157643,5.142765,5.155709e+05,6.0,73.0,14114.000000
std,181.432265,3.947166e+04,7.473292,2882.336492,2038.164080,1404.572326,2882.336492,302.545443,74.102849,30463.047753,0.055711,951.031552,632.039961,1.084002,2.677185e+05,0.0,0.0,39471.655779
min,0.000000,6.073000e+09,16.500000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,22614.000000,0.283000,0.000000,0.000000,2.300000,1.790000e+04,6.0,73.0,100.000000
25%,156.750000,6.073008e+09,32.400000,3684.750000,2556.250000,1609.250000,3684.750000,354.750000,13.000000,52725.500000,0.374475,1295.250000,373.250000,4.200000,3.592000e+05,6.0,73.0,8335.750000
50%,313.500000,6.073013e+09,36.200000,4820.000000,3372.000000,2161.000000,4820.000000,496.000000,41.000000,70189.500000,0.407500,1750.000000,698.000000,5.142765,4.544500e+05,6.0,73.0,13409.500000
75%,470.250000,6.073018e+09,41.300000,6218.500000,4622.250000,2843.750000,6218.500000,675.750000,97.250000,92343.750000,0.440550,2307.250000,1109.500000,5.900000,5.974000e+05,6.0,73.0,17701.250000
max,627.000000,6.073990e+09,72.500000,40402.000000,30455.000000,24143.000000,40402.000000,3180.000000,578.000000,191642.000000,0.666700,9145.000000,7146.000000,7.900000,2.000001e+06,6.0,73.0,990100.000000



--- Possible geographic ID columns ---
['GEOID', 'geometry', 'tract']


## San Diego Census 2017 (ACS)

**Datasets explored**
- San Diego.csv: 628 rows x 21 columns - economic, social, geographic per census tract
- metaindicator.csv: column dictionary for 13 key indicators

**Data quality**
- 628 tracts covered (matches San Diego County's expected count)
- 0 missing values across all columns (cleaner than the pedestrian data)
- Tract IDs in GEOID column
- Geometry column included as WKT POLYGONs - no separate shapefile needed for mapping
- One labeling inconsistency: total_bachelor column has "No schooling completed" label, needs verification

**Distributions (key insights)**
- Population per tract: mean 5,228, median 4,820, max 40,402 (high variance suggests dense vs sparse zones)
- Median age: mean 37, range 16.5-72.5
- 
**Next steps**
- Verify CRS of geometry column (looks like Web Mercator / EPSG:3857 based on coordinate magnitudes)
- Spatial join census tracts with MTS stops to count stops per tract
- Cross-reference high pedestrian counts with high-population tracts
- Identify low-stop-density / high-population tracts as underserved candidates
- Investigate the total_bachelor column labeling issue